include/interfaces/hydrogels_one_field.h

```diff
@@ -0,0 +1,244 @@
+#include <deal.II/lac/solver_gmres.h>
+
+#include "lac/lac_type.h"
+
+
+template <int dim, int spacedim, typename LAC>
+class HydroGelOneField : public PDESystemInterface<dim,spacedim, HydroGelOneField<dim,spacedim,LAC>, LAC>
+{
+public:
+
+  ~HydroGelOneField() {};
+
+  HydroGelOneField();
+
+  void declare_parameters (ParameterHandler &prm);
+  void parse_parameters_call_back ();
+
+//  void set_matrix_couplings(std::vector<std::string> &couplings) const;
+
+  template <typename EnergyType, typename ResidualType>
+  void energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                              FEValuesCache<dim,spacedim> &scratch,
+                              std::vector<EnergyType> &energies,
+                              std::vector<std::vector<ResidualType> > &local_residuals,
+                              bool compute_only_system_terms) const;
+
+
+  /*void compute_system_operators(const DoFHandler<dim,spacedim> &,
+                                const std::vector<shared_ptr<LATrilinos::BlockMatrix> >,
+                                LinearOperator<LATrilinos::VectorType> &,
+                                LinearOperator<LATrilinos::VectorType> &) const;
+  */
+
+
+private:
+  double T;
+  double Omega;
+  double G;
+  double chi;
+  double l0;
+
+  double mu0;
+  double l02;
+  double l03;
+  double l0_3;
+  double l0_6;
+  const double R=8.314;
+
+
+  /*  mutable shared_ptr<TrilinosWrappers::PreconditionAMG> U_prec;
+    mutable shared_ptr<TrilinosWrappers::PreconditionJacobi> p_prec;
+    mutable shared_ptr<TrilinosWrappers::PreconditionJacobi> c_prec;
+  */
+
+};
+
+template <int dim, int spacedim, typename LAC>
+HydroGelOneField<dim,spacedim,LAC>::HydroGelOneField() :
+  PDESystemInterface<dim,spacedim,HydroGelOneField<dim,spacedim,LAC>, LAC>("Free Swelling One Field",
+      dim,1,
+      "FESystem[FE_Q(2)^d]",
+      "u,u,u","1")
+{}
+
+/*
+template <int dim, int spacedim, typename LAC>
+void
+HydroGelOneField<dim,spacedim,LAC>::
+set_matrix_couplings(std::vector<std::string> &couplings) const
+{
+  couplings[0] = "1,1,0;1,0,1;0,1,1";
+  couplings[1] = "0,0,0;0,1,0;0,0,0";
+}
+*/
+
+template <int dim, int spacedim, typename LAC>
+template <typename EnergyType, typename ResidualType>
+void
+HydroGelOneField<dim,spacedim,LAC>::
+energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                       FEValuesCache<dim,spacedim> &fe_cache,
+                       std::vector<EnergyType> &energies,
+                       std::vector<std::vector<ResidualType> > &,
+                       bool compute_only_system_terms) const
+{
+  EnergyType alpha = this->alpha;
+  this->reinit(alpha, cell, fe_cache);
+
+  const FEValuesExtractors::Vector displacement(0);
+
+  auto &Fs = fe_cache.get_deformation_gradients("solution", "Fu", displacement, alpha);
+
+  auto &JxW = fe_cache.get_JxW_values();
+
+  for (unsigned int q=0; q<n_q_points; ++q)
```

Comment:
Grazie mille, avevo visto che in altre interfacce in pi-domus usate size() per ricavare il n di pti di quadratura. Cmq non compila ;-) C'è un problema con l'overloading di = nel seguente blocco di codice:
```
Tensor<2,dim,ResidualType> F_star = J*transpose(invert(F_res));
auto grad_v = fev[displacement].gradient(i,q);
residuals[0] -= mu0*(this->t)*F_star*grad_v;
```